tests/testVariance.py

```diff
@@ -127,6 +127,20 @@ def testVariance(self):
 
         self.assertTrue(np.abs(source.get("base_Variance_value") - variance) < varianceStd)
 
+        # flag_emptyFootprint should not have been set since the footprint has non-masked pixels at this
+        # point.
+        self.assertFalse(source.get("base_Variance_flag_emptyFootprint"))
+
+        # Now set the pixel mask for all pixels to 'BAD' and remeasure.
+        mask.getArray()[:,:] = mask.getPlaneBitMask("BAD")
+        task.run(catalog, exp)
+
+        # The computed variance should be nan and flag_emptyFootprint should have been set since the footprint
+        #has all masked pixels at this point.
+        self.assertTrue(np.isnan(source.get("base_Variance_value")))
+        self.assertTrue(source.get("base_Variance_flag_emptyFootprint"))
+
```

Comment:
If it were possible, it would be nice to make this a separate test method -- it gives us more granularity when detecting failures. You could put all the logic which sets up and runs the test in common function and then have two tests which call it.